test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -326,6 +326,25 @@ object Assertion {
       assertion.test(actual.size)
     }
 
+  final def hasFirst[A](assertion: Assertion[A]): Assertion[Iterable[A]] =
```

Comment:
When you construct an assertion that takes another assertion as a parameter, you want to use `assertionRec` so that the sub-assertion and any further sub-assertions it contains are recursively embedded in the assertion result so we get nice hierarchical reporting of test failures.

Reply:
Switched to `assertionRec`